### PR TITLE
docs: thoroughly document DataTypes

### PR DESCRIPTION
**Pull request by ephys**
This PR rewrites the other-data-types page to thoroughly document every built-in DataType, and their corresponding SQL type (per dialect).

I based it on the DataTypes unit tests, as well as the documentation of the different dialects.
I'm unsure which documentation should be used for db2 & ibmi so there are likely errors in what I wrote for these 2 dialects. @markdirish if you have a minute to review this part, and/or point me to the right resources.

I've also documented common pain points, such as DataTypes.UUIDV4 being a JS function instead of a SQL one.

I've also moved the part about querying JSON to the page about Querying (with a link leading there), as that seems to be the place where people look for it.

I wrote a react component that filters out the columns of a table based on the user's preferred dialect :)

image

image

### CODE REVIEW

**Comment by ephys** [comment]
I'll fix the lint issue but feel free to review the contents :)

File: docs/core-concepts/model-basics.md [hunk 1]
```diff
@@ -308,18 +308,22 @@ sequelize.define('User', {
...
 
 ```js
 sequelize.define('Foo', {
-  bar: {
-    type: DataTypes.DATETIME,
-    defaultValue: DataTypes.NOW
-    // This way, the current date/time will be used to populate this column (at the moment of insertion)
+  myUuid: {
+    type: DataTypes.UUID,
+    defaultValue: fn('uuid_generate_v4'),
```
Comment (sdepold) [inline docs/core-concepts/model-basics.md:317]: omg this is amazing :o

**Review by sdepold — COMMENTED** [review]
What are the React files for?

**Comment by sdepold** [comment]
> What are the React files for?

Ah nevermind. Cool stuff

**Review by sdepold — DISMISSED** [review]

**Comment by ephys** [comment]
> What are the React files for?

They're two 2 components that can be used in mdx files: 

- `Deprecated` adds a 'Trash' icon next to a piece of text, & hovering the text display "this feature is deprecated"
  image
- `DialectTableFilter` is used to make the support tables more readable by filtering out the dialects that don't concern you

File: docs/core-concepts/model-basics.md (hunk 1 above)
Comment (ephys) [reply docs/core-concepts/model-basics.md:317]: confirmed 
image

**Comment by ephys** [comment]
Before merging I've noticed the two first datatype sections don't have links to the dialect documentations like the other ones. I should fix that.

**Comment by ephys** [comment]
Links added

IMO we can merge without @markdirish's review for db2/ibmi. We'll open a follow-up PR if necessary

**Comment by sdepold** [comment]
lgtm

**Comment by ephys** [comment]
conflict fixed

**Review by sdepold — APPROVED** [review]

**Pull request merged**